[tayruxin] iP

tayruxin opened the pull request:
**DukePro**
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's, 

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is, 

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😙

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method: 

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tayruxin committed:
Implement Level-0: Rename, Greet, Exit

tayruxin committed:
Implement Level 1: Echo

tayruxin committed:
Implement Level 2: Add, List

tayruxin committed:
Implement Level-3: Mark as Done

tayruxin committed:
Implement Level 4: ToDos, Events, Deadlines

tayruxin committed:
Implement A-TextUiTesting: Automated Text UI Testing

tayruxin committed:
Implement Level 5: Handle Errors

tayruxin committed:
Implement Level-6: Delete

tayruxin committed:
Implement Level-7: Save

tayruxin committed:
Merge branch-Level-7

tayruxin committed:
Implement Level-8: Dates and Time

tayruxin committed:
Merge branch-Level-8

tayruxin committed:
Implement A-MoreOOP

tayruxin committed:
Merge branch-A-MoreOOP into master

tayruxin committed:
Implement A-Packages

tayruxin committed:
Merge add-gradle-support

tayruxin committed:
Implement A-Gradle

tayruxin committed:
Implement A-JUnit: Add JUnit Tests

tayruxin committed:
Implement A-jar: Create a JAR file

tayruxin committed:
Implement A-JavaDoc: Add JavaDoc comments

tayruxin committed:
Implement A-CodingStandard

tayruxin committed:
Implement Level-9: Find

tayruxin committed:
Merge branch-A-CodingStandard

tayruxin committed:
Merge branch-A-CodingStandard

tayruxin committed:
Add number label to the matching list for find feature

tayruxin committed:
Implement level-10

tayruxin committed:
Change duke.txt data

tayruxin committed:
Merge branch-Level-10

Cikguseven commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,106 @@
+package duke;
+import java.time.LocalDate;
+import java.time.LocalTime;
```
Maybe you can consider using LocalDateTime to store both date and time data?

Cikguseven commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,106 @@
+     * @param storage storage class that handles the loading and saving of files
+     * @throws DukeException if the input does not much the required format
+     */
+    public static void parseInput(String input, TaskList list, int number,
```
Perhaps you can format your input in a case-insensitive manner so that you do not have to call .equalsIgnoreCase later on?

Cikguseven commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+        this.time = time;
+    }
+
+    /**
+     * Return the string representation of deadline in the right date and time format
+     *
+     * @return a string with the right date and time format
+     */
```
Perhaps you can add punctuation to the end of Javadoc descriptions and parameters to comply to the coding standard?

Cikguseven commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,44 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DeadlineTest {
```
Good job on having multiple non-trivial test cases for the various tasks!

Cikguseven reviewed (commented):
Code has logical flow and is easy to read. You can consider placing similar classes in separate packages, such as Task related classes.

tayruxin committed:
Edit the text box to accomodate to the scroll pane

RiyaMehta2211 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,206 @@
+//CHECKSTYLE.OFF: MissingJavadocMethodCheck
+
+/**
+ * Class that is responsible for handling the input of the user
```
Good job giving class header comments to explain the functions of the particular class

RiyaMehta2211 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,206 @@
+        } else if (input.startsWith("delete")) {
+            number = 1;
+        } else if (input.equalsIgnoreCase("todo")) {
+            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
```
Great job on general encapsulation of different exceptions under a general DukeException!

RiyaMehta2211 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,206 @@
+                    String[] parts = date.split(" ");
+                    String dateString = parts[0];
+                    String timeString = parts[1];
+                    d1 = LocalDate.parse(dateString, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
```
Perhaps you can consider catching java.time.format.DateTimeParseException here for invalid inputs that cannot be parsed to extract the date or time :)

RiyaMehta2211 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,206 @@
+                }
+                list.add(deadline);
+                number++;
+                System.out.println("Got it. I've added this task:\n" + deadline.toString());
```
perhaps transfer these print messages under functions in Ui.java as the Ui class involves handling chatbot's interactions with users

RiyaMehta2211 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,206 @@
+
+
+    /**
+     * Read the input from the user and carry out the required action for Gui
```
Great job on the detailed javadocs. clearly explaining the different parameters and the exceptions being thrown in the function

RiyaMehta2211 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,206 @@
+            } else {
+                throw new DukeException("Invalid input format.");
+            }
+        } else if (input.startsWith("event")) {
```
Great job on abiding by the coding standards (eg. appropriate indentation levels, and egyptian style braces being used!)

RiyaMehta2211 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,71 @@
+package duke;
+import java.io.File;
```
great job ensuring no wildcard imports were used!

RiyaMehta2211 reviewed (commented):
Overall I found the code to be well in line with the OOP principles with some minor changes suggested in view of the classes being made and their purposes. The coding standards were followed well, with appropriate indentation levels. There were no wildcard imports used and egyptian style bracing was used. Good job!

tayruxin committed:
Implement A-Assertions

Code does not have assert function. Assert function can be useful for debugging. Let's add assert functions to Task, Parser and Storage class

tayruxin committed:
Implement A-CodeQuality

tayruxin committed:
Merge pull request #2 from tayruxin/branch-A-Assertions

Implement A-Assertions

tayruxin committed:
Merge branch 'master' into branch-A-CodeQuality

tayruxin committed:
Merge pull request #1 from tayruxin/branch-A-CodeQuality

Implement A-CodeQuality

tayruxin committed:
Implement A-CI

tayruxin committed:
Delete lines that are not needed in gradle.yml

tayruxin committed:
Remove input and output check

tayruxin committed:
Remove irrelevant platforms from gradle.yml

tayruxin committed:
Correct the styling issues from the various test files

tayruxin committed:
Implement C-Priority

tayruxin committed:
Insert Ui.png

tayruxin committed:
Implement A-UserGuide

tayruxin committed:
Include hyperlinks in the user guide

tayruxin committed:
Edit javadocs formatting

tayruxin committed:
Remove not needed System.out.print

tayruxin committed:
Insert a command summary in the user guide

tayruxin committed:
Remove table summary
